config/routes.rb

```diff
@@ -7,4 +7,7 @@
   resources :users do
     resources :posts
   end
+  get "/users/:id/posts/:id/comments/new", to: "comments#new", as: 'new_comment'
+  post "/users/:id/posts/:id/comments", to: "comments#create", as: 'comments'
+  post "/users/:id/posts/:id", to: "likes#create", as: 'likes'
 end
```

Comment:
- [ ] Your routes in line 10 - 12 are not properly defined. They should have `users/:user_id`, `posts/:post_id` kind of context as is the norm in Rails. In your controllers, if we select `params[:id]` we cannot really tell which `:id` is being picked since there are multiple `:id`s in the route. You can fix this by doing this:

  ```
  resources :users do
    resources :posts do
      resources :comments, only: [:new, :create]
      resources :likes, only: [:create]
    end
  end
  ```

  The routes above will help us with params like `:post_id` and `user_id` if we need them and as required.